include/flamegpu/util/StringPair.h

```diff
@@ -14,26 +14,44 @@ namespace util {
 typedef std::pair<std::string, std::string> StringPair;
 
 /**
- * Hash function so that StringPair can be used as a key in a map
+ * Compare function so that StringPair can be used in a map
  */
-struct StringPairHash {
-    size_t operator()(const std::pair<std::string, std::string>& k) const {
-        return std::hash<std::string>()(k.first) ^
-            (std::hash<std::string>()(k.second) << 1);
+template<typename T1, typename T2>
```

Comment:
Techinically these changes might be an api break, even though its a bit no one is intended to directly use. 

Might be worth mentioning but not in much detail in a cahngelog?

Reply:
It's not a breaking change.

`StringPairHash` is redefined below as a `typedef`, I just made the implementation more generalised via templating.